spras/analysis/summary.py

```diff
@@ -44,28 +47,44 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame) ->
         number_nodes = nw.number_of_nodes()
         number_edges = nw.number_of_edges()
         ncc = nx.number_connected_components(nw)
+
         # Initialize list to store current network information
         cur_nw_info = [nw_name, number_nodes, number_edges, ncc]
+
         # Iterate through each node property and save the intersection with the current network
         for node_list in nodes_by_col:
             num_nodes = len(set(nw).intersection(node_list))
             cur_nw_info.append(num_nodes)
+
+        # String split name to access algorithm and hashcode from filepath
+        # Name of filepath follows format "output/data#-algo-params-hashcode/pathway.txt"
+        # algorithm parameters have format { algo : { hashcode : { parameter combos } } }
+
+        filepath = nw_name.split("/")
+        filename = filepath[1].split("-")
```

Comment:
Will this work for outputs stored in subfolders within the main output folder?

example `output/example/data#-algo-params-hashcode/pathway.txt"`

Reply:
Can get the list of algorithm-params-hashcode from `algorithms_with_params` in the Snakefile and then pass it to the function so you don't have to extract it